### PR TITLE
Properly format os.version property

### PR DESCRIPTION
**Pull request by gblues**
=== ISSUE
fix for issue #5 
=== DETAILS
AIX is a weird beast, where the OS release and version are stored in
separate parts of the `utsname` structure.

By default, the JVM only grabs one of the two fields. This is what
causes `os.version` to return `2` on AIX 7.2.

I've added a function that allocates a string and writes the formatted
version into it. Since AIX' `strdup()` implementation performs a
`malloc()`, this effectively preserves the existing behavior.
=== TESTING
Unfortunately, I don't have an AIX system to do any kind of a test build
on.

Potential pitfalls:

* The method's been added to the wrong file.
* The AIX build has neither `__linux__` or `_ALLBSD_SOURCE` defined,
  and thus compilation fails due to missing `stdio.h` include
* dumb human error

### CODE REVIEW

**Comment by karianna** [comment]
@gblues - Have you reported this bug in the upstream OpenJDK 8u project?  I think we should test your fix here and then report the bug along with this associated patch in OpenJDK proper.  Only if they refuse to accept the patch should we then apply it here (and in 9 etc).

File: jdk/src/solaris/native/java/lang/java_props_md.c [hunk 1]
```diff
@@ -380,6 +380,24 @@ static char* getEmbeddedToolkit() {
 }
 #endif
 
+#ifdef AIX
+char *
```
Comment (karianna) [inline jdk/src/solaris/native/java/lang/java_props_md.c]: Not sure why that char * is there?

File: jdk/src/solaris/native/java/lang/java_props_md.c [hunk 2]
```diff
@@ -380,6 +380,24 @@ static char* getEmbeddedToolkit() {
...
+GetAixVersionString(char *version, char *release)
```
Comment (karianna) [inline jdk/src/solaris/native/java/lang/java_props_md.c]: start with Camel case

File: jdk/src/solaris/native/java/lang/java_props_md.c [hunk 3]
```diff
@@ -380,6 +380,24 @@ static char* getEmbeddedToolkit() {
...
+{
+    char *result = NULL:
+    // the 2 accounts for null-terminator and the '.' between release and version strings
```
Comment (karianna) [inline jdk/src/solaris/native/java/lang/java_props_md.c]: Can you edit the comment with an example of what this looks like?

File: jdk/src/solaris/native/java/lang/java_props_md.c [hunk 4]
```diff
@@ -380,6 +380,24 @@ static char* getEmbeddedToolkit() {
...
+    int bytesNeeded = strlen(release) + strlen(version) + (2 * sizeof(char))
+
+    // on AIX, strdup() invokes malloc
+    result = (char *)malloc(bytesNeeded);
```
Comment (karianna) [inline jdk/src/solaris/native/java/lang/java_props_md.c]: We've malloc'd, do we need to release anything once we're done?

**Review by karianna — CHANGES_REQUESTED** [review]
A few Q's and comments

File: jdk/src/solaris/native/java/lang/java_props_md.c (hunk 4 above)
Comment (gblues) [reply jdk/src/solaris/native/java/lang/java_props_md.c]: No. `GetAixVersionString` follows the same semantics as `strdup`, in that it is the caller's responsibility to free the memory. In this way it is drop-in compatible with the previous code which simply `strdup()`ed the version field out of the uname structure.

File: jdk/src/solaris/native/java/lang/java_props_md.c (hunk 3 above)
Comment (gblues) [reply jdk/src/solaris/native/java/lang/java_props_md.c]: No problem. I've just pushed up an edit (and caught a couple of my own dumb errors too).

**Comment by gblues** [comment]
@karianna it looks like it's already been resolved in OpenJDK 9 (https://bugs.openjdk.java.net/browse/JDK-8132374).

I don't have JIRA access to directly request a backport, but I've e-mailed the jdk8u lead (Sean Coffey).

File: jdk/src/solaris/native/java/lang/java_props_md.c (hunk 4 above)
Comment (karianna) [reply jdk/src/solaris/native/java/lang/java_props_md.c]: Perfect, thanks

**Comment by karianna** [comment]
OK, let's see what Sean says.  This patch is now good, but I'd prefer (no offense!) just to have the OpenJDK 9 one backported (consistency).

**Review by karianna — APPROVED** [review]

**Comment by gblues** [comment]
Haha, no offense taken!

Latest news from upstream: Sean has reached out to Volker Simonis, who's just put a RFA out to jdk8u-dev.

**Comment by gblues** [comment]
@karianna Upstream has backported the JDK9 fix: https://bugs.openjdk.java.net/browse/JDK-8187859. Closing.

**Comment by karianna** [comment]
@gblues Great work!

**Pull request closed without merging**